java/fury-core/src/main/java/org/apache/fury/serializer/ObjectSerializer.java

```diff
@@ -228,6 +229,65 @@ public void write(MemoryBuffer buffer, T value) {
     writeContainerFields(buffer, value, fury, refResolver, classResolver);
   }
 
+  @Override
+  public T copy(T originObj) {
+    if (isRecord) {
+      return originObj;
```

Comment:
fields of Record may be mutable, we should not skip its update unless we can ensure all fields are immutable types. And if all fields value are immutable, and fields are final, we can take it as immutable even if it's not record